### PR TITLE
{math}[foss/2019b,fosscuda/2019b] Keras v2.3.1, TensorFlow v2.1.0, libgpuarray v0.7.6, Theano v1.0.4, R-keras v2.2.5.0 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by edmondac**
(created using `eb --new-pr`)

This depends on ~~https://github.com/easybuilders/easybuild-easyconfigs/pull/9519~~ being merged first

### CODE REVIEW

**Comment by edmondac** [comment]
That patch is to make Bazel compile on my Ubuntu 19.10 machine. It comes from https://github.com/clearlinux-pkgs/bazel/blob/adefd9046582cb52f39579033132e6265ef6ddb0/rename-gettid-functions.patch

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 9 out of 11 (8 easyconfigs in this PR)
bear-pg0212u17a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/dea4ee1953a139a3e57265d920e0bb58 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 5 out of 8 (8 easyconfigs in this PR)
bear-pg0212u15b.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/2452abd1436f51c35cf8ef721bcb6d17 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 8 (8 easyconfigs in this PR)
bear-pg0212u15b.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/849e2f12c6cbd9720691a4f6cf704368 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
bear-pg0212u15b.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/8db204ea8acc31f9802b7f88b4ca6a15 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 112 out of 121 (8 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/b223b11c261c251735f603a2db213ab4 for a full test report.

**Comment by edmondac** [comment]
> Test report by @terjekv
> **FAILED**
> Build succeeded for 112 out of 121 (8 easyconfigs in this PR)
> ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
> See https://gist.github.com/b223b11c261c251735f603a2db213ab4 for a full test report.

I think that's due to a missing Java 1.8 source file which needs to be downloaded manually

**Comment by terjekv** [comment]
> > Test report by @terjekv
> > **FAILED**
> > Build succeeded for 112 out of 121 (8 easyconfigs in this PR)
> > ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
> > See https://gist.github.com/b223b11c261c251735f603a2db213ab4 for a full test report.
> 
> I think that's due to a missing Java 1.8 source file which needs to be downloaded manually

Correct. I had the wrong version of jdk in the sources directory. I'm restarting the build now, but from the point of failure.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 9 out of 9 (8 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/9b0f066ce025dc064281bc25e2e11781 for a full test report.

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+builddependencies = [('CMake', '3.15.3')]
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('SciPy-bundle', '2019.10', versionsuffix),
+    ('Mako', '1.1.0'),
+    ('NCCL', '2.4.8'),
+]
+
+configopts = '-DCMAKE_BUILD_TYPE=Release'
```
Comment (Flamefire) [inline easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b-Python-3.7.4.eb]:
```suggestion
build_type = 'Release'
```

see https://github.com/easybuilders/easybuild-easyblocks/blob/85c2f77c2243d244fa21eaac40de960018e3aa3f/easybuild/easyblocks/generic/cmakemake.py#L76

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    'source_tmpl': '%(name)s_%(version)s.tar.gz',
+}
+exts_list = [
+    ('config', '0.3', {
+        'checksums': ['5ade6a768db6225634095ac0e71e5710fd1597c15512bac0b45a0b52afb3c750'],
+    }),
+    ('tfruns', '1.4', {
+        'checksums': ['6fc31f3b75c1888d3d14fe14a31c3567dab1bc2a4d0a61c0c72d543c422d1bd7'],
+    }),
+    ('tensorflow', '2.0.0', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb]: Please use 2.1

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.0.0-foss-2019b-Python-3.7.4.eb [hunk 3]
```diff
@@ -0,0 +1,142 @@
+easyblock = 'PythonBundle'
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.0.0-foss-2019b-Python-3.7.4.eb]: Please no more 2.0

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb (hunk 2 above)
Comment (edmondac) [reply easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb]: There isn't a cran package for 2.1.0 - but the existing cran 2.0.0 should work with a tf 2.1.0 backend, I hope...

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.0.0-foss-2019b-Python-3.7.4.eb (hunk 3 above)
Comment (edmondac) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.0.0-foss-2019b-Python-3.7.4.eb]: Done

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb (hunk 2 above)
Comment (Flamefire) [reply easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb]: Oh so this is not actually the TF version used?

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    'source_tmpl': '%(name)s_%(version)s.tar.gz',
+}
+exts_list = [
+    ('config', '0.3', {
+        'checksums': ['5ade6a768db6225634095ac0e71e5710fd1597c15512bac0b45a0b52afb3c750'],
+    }),
+    ('tfruns', '1.4', {
+        'checksums': ['6fc31f3b75c1888d3d14fe14a31c3567dab1bc2a4d0a61c0c72d543c422d1bd7'],
+    }),
+    ('tensorflow', '2.0.0', {
```
[the remark this answers is not in the document]
Comment (edmondac) [reply easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb]: That's a cran package - and there isn't a 2.1.0. I'm hoping the 2.0.0 cran package works with the python tf 2.1.0

**Comment by edmondac** [comment]
Since the resolved discussions hide things, I'll put this here:

In `R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb` we have:
```
('tensorflow', '2.0.0', {
```

That's a cran package - and there isn't a 2.1.0. I'm hoping the 2.0.0 cran package works with the python tf 2.1.0

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 12 (8 easyconfigs in this PR)
bear-pg0211u03a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9727a9627f28ece9d70fb27564aaeb96 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 9 (8 easyconfigs in this PR)
bear-pg0212u15b.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/d79b4352016f0cb44ea94e2aa0b83626 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 9 out of 9 (8 easyconfigs in this PR)
bear-pg0212u15b.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/dd0cce43fab4417b28bffde72ddc3dc5 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 6 out of 10 (8 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/9015e99724c9d6f94f498f263ac505cd for a full test report.

**Comment by edmondac** [comment]
> Test report by @terjekv
> **FAILED**
> Build succeeded for 6 out of 10 (8 easyconfigs in this PR)
> ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
> See https://gist.github.com/9015e99724c9d6f94f498f263ac505cd for a full test report.

Those errors are weird... you might need to shoot it and try the test report again!

**Comment by terjekv** [comment]
> > Test report by @terjekv
> > **FAILED**
> > Build succeeded for 6 out of 10 (8 easyconfigs in this PR)
> > ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
> > See https://gist.github.com/9015e99724c9d6f94f498f263ac505cd for a full test report.
> 
> Those errors are weird... you might need to shoot it and try the test report again!

I think a cleanup job may have clobbered the build. I'll try restarting it.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 10 out of 10 (8 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/e5b7a4a8821d37c79fca2ec37572a7cd for a full test report.

**Comment by edmondac** [comment]
> Looks good to me.

Hooray! :-)

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb (hunk 2 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb]: A slight delay here, but no, the R extension named "TensorFlow" is a binding library that mostly tracks TensorFlow proper. Mostly.

**Comment by Flamefire** [comment]
Doesn't work for me as `pip check` fails: TF 2.1 has a newer scipy but loading the other modules overwrites this with the SciPyBundle containing the older SciPy. I'd suggest putting TF down in the list

**Comment by edmondac** [comment]
> Doesn't work for me as `pip check` fails: TF 2.1 has a newer scipy but loading the other modules overwrites this with the SciPyBundle containing the older SciPy. I'd suggest putting TF down in the list

Would you mind adding a test report @Flamefire ?

**Comment by Flamefire** [comment]
It is going to fail with
> FAILED: Installation ended unsuccessfully (build directory: /tmp/s3248973-EasyBuild/Keras/2.3.1/fosscuda-2019b-Python-3.7.4): build failed (first 300 chars): cmd "pip check" exited with exit code 1 and output:
tensorflow 2.1.0 has requirement scipy==1.4.1; python_version >= "3", but you have scipy 1.3.1.

due to the above mentioned ordering problem. I'm surprised this works for you. Are you using latest EB? Can you load the Theano build by this and run `pip check`?

**Comment by edmondac** [comment]
I've cleared some things out and I'm running a new test report now...

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 12 (8 easyconfigs in this PR)
bear-pg0211u03a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a7364787dae82148823152758a2aff1d for a full test report.

**Comment by Flamefire** [comment]
Would you mind factoring out the R deps into another PR? I can't test this in our Power9 system because R won't install due to missing Java 11

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 6 out of 12 (8 easyconfigs in this PR)
bear-pg0211u03a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9fa6ddc68bedb112740063757048a445 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 12 (8 easyconfigs in this PR)
bear-pg0211u03a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz, Python 2.7.5
See https://gist.github.com/0da5b53c21f83140dfeb51eafbb612e9 for a full test report.

**Comment by edmondac** [comment]
> Would you mind factoring out the R deps into another PR? I can't test this in our Power9 system because R won't install due to missing Java 11

https://github.com/easybuilders/easybuild-easyconfigs/pull/9371 is now merged, with Java 11 for POWER9

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 8 (8 easyconfigs in this PR)
bber0501u03a.bb2.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/c4dbd8813599d9024b160a762a6f46ed for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 121 out of 123 (8 easyconfigs in this PR)
bber0501u03a.bb2.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/9ef81f3992832db3d78686e5b5f3111e for a full test report.

**Comment by terjekv** [comment]
@edmondac 

```
Error: package or namespace load failed for liquidSVM in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/rds/bear-apps/devel/2019b/edmondac-eb-2/EL7/EL7-haswell/software/R/3.6.2-fosscuda-2019b/lib64/R/library/00LOCK-liquidSVM/00new/liquidSVM/libs/liquidSVM.so':
  libcuda.so.1: cannot open shared object file: No such file or directory
```

That feels awfully familiar. That's from the OS driver package, isn't it?

**Comment by Flamefire** [comment]
Actually liquidSVM should be removed. See discussion https://github.com/easybuilders/easybuild-easyconfigs/pull/9544/files#r365531097

**Comment by edmondac** [comment]
Yes liquidSM should be removed, I agree.

But my error was trying to build fossucda things on a box without a GPU (or CUDA drivers installed). Doh!

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 9 (8 easyconfigs in this PR)
bear-pg0212u15a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/4508ae62bc33cd29c9ad35338754126b for a full test report.

**Comment by Flamefire** [comment]
Seems it is not so reasonable to remove liquidSVM. It was "recently" updated on CRAN and it is failing due to this on nvcc in GCCcore/8.3.0/include/c++/8.3.0/type_traits:

```
#if !defined(__STRICT_ANSI__) && defined(_GLIBCXX_USE_FLOAT128)
   template<>
      struct __is_floating_point_helper<__float128>
     : public true_type { };
#endif
```

Some more links on that: https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1717257, or even https://github.com/migueldiascosta/easybuild-easyconfigs/commit/480a4df8bf383975f76b86450a90503f5cb7afb6#diff-9d734c9069d6eb20c2678afc2959918aR65-R66

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 10 out of 10 (8 easyconfigs in this PR)
bear-pg0212u15a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/4339af8d3ceac1c7a5ec3938642d03d1 for a full test report.

**Comment by edmondac** [comment]
> Seems it is not so reasonable to remove liquidSVM. It was "recently" updated on CRAN and it is failing due to this on nvcc in GCCcore/8.3.0/include/c++/8.3.0/type_traits:
> 
> ```
> #if !defined(__STRICT_ANSI__) && defined(_GLIBCXX_USE_FLOAT128)
>    template<>
>       struct __is_floating_point_helper<__float128>
>      : public true_type { };
> #endif
> ```
> 
> Some more links on that: https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1717257, or even [migueldiascosta@480a4df#diff-9d734c9069d6eb20c2678afc2959918aR65-R66](https://github.com/migueldiascosta/easybuild-easyconfigs/commit/480a4df8bf383975f76b86450a90503f5cb7afb6#diff-9d734c9069d6eb20c2678afc2959918aR65-R66)

Can we shift liquidSVM discussions to https://github.com/easybuilders/easybuild-easyconfigs/pull/9544 ? Then we might be able to get this one merged!

**Comment by Flamefire** [comment]
> Can we shift liquidSVM discussions to #9544 ? Then we might be able to get this one merged!

Sure but until then I can't test on Power9. I also don't see how your pip check can succeed if Theano is loaded after TensorFlow and Theano loads a new SciPy which clashes with the TF scipy. I'll rerun a build though.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 6 out of 8 (8 easyconfigs in this PR)
taurusi5398.taurus.hrsk.tu-dresden.de - Linux RHEL 7.4, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/333dbb662eafe48947985f9aa3ce295b for a full test report.

**Comment by edmondac** [comment]
Theano says: `Requires-Dist: scipy (>=0.14)`

The SciPy bundle in there has `1.3.1`

TensorFlow installs `1.4.1`

`pip check` runs ok with these modules loaded

TensorFlow depends on h5py, which itself depends on SciPy-bundle 2019.10 - so that SciPy will always be there when TensorFlow is loaded and so the new scipy in tensorflow will always take over.

Theano depends on libgpuarray 0.7.6, which depends on SciPy-bundle 2019.10.

Whichever of TensorFlow and Theano is loaded first, the SciPy-bundle module will be loaded *before* TensorFlow, and so the new scipy in the TensorFlow eb will always take precedence...

**Comment by Flamefire** [comment]
> Whichever of TensorFlow and Theano is loaded first, the SciPy-bundle module will be loaded before TensorFlow, and so the new scipy in the TensorFlow eb will always take precedence...

(At least on our system) loading an already loaded module reloads it. I believe this is due to a line `conflict("SciPy-bundle")` in the lua file. Hence what happens when you load e.g. Keras is:

- ml TensorFlow
  - ml SciPy -> Prepend to PYTHONPATH
  - Prepend TF scipy to PYTHONPATH
- ml Theano
  - ml SciPy -> Unload Scipy, Reload, prepend to PYTHONPATH

--> the sciypy from SciPy(-Bundle) is *before* the TF scipy in PYTHONPATH and hence gets used leading to the failure in my test report

**Comment by edmondac** [comment]
> > Whichever of TensorFlow and Theano is loaded first, the SciPy-bundle module will be loaded before TensorFlow, and so the new scipy in the TensorFlow eb will always take precedence...
> 
> (At least on our system) loading an already loaded module reloads it. I believe this is due to a line `conflict("SciPy-bundle")` in the lua file. Hence what happens when you load e.g. Keras is:
> 
>     * ml TensorFlow
>       
>       * ml SciPy -> Prepend to PYTHONPATH
>       * Prepend TF scipy to PYTHONPATH
> 
>     * ml Theano
>       
>       * ml SciPy -> Unload Scipy, Reload, prepend to PYTHONPATH
> 
> 
> --> the sciypy from SciPy(-Bundle) is _before_ the TF scipy in PYTHONPATH and hence gets used leading to the failure in my test report

I've not heard of that behaviour before - the system's I've used have always only loaded modules that aren't already loaded. Maybe this is one for the slack channel to see if what other people's systems would do?

**Comment by edmondac** [comment]
OK - I've re-ordered them since it seems either behaviour is possible

**Comment by edmondac** [comment]
Test report by @bear-rsg
**FAILED**
Build succeeded for 2 out of 8 (8 easyconfigs in this PR)
bear-pg0212u15a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/8b3d3584aa58486313c102f0117e74a8 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 10 out of 10 (8 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/2faeb8881f5a76da5642a0d65ab7aea3 for a full test report.

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 9 out of 9 (8 easyconfigs in this PR)
bear-pg0212u15a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/b6f8e6181b7d47323825530f3d34e6f0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 9 out of 11 (8 easyconfigs in this PR)
node3170.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/e25f0dc36f832b6b6af19edd74efb278 for a full test report.

**Comment by terjekv** [comment]
> Test report by @boegel
> **FAILED**
> Build succeeded for 9 out of 11 (8 easyconfigs in this PR)
> node3170.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
> See https://gist.github.com/e25f0dc36f832b6b6af19edd74efb278 for a full test report.

Checksum failure for R packages? Well I never... Shocking!

**Comment by Flamefire** [comment]
Can we have an `--add-alternate-checksum` option to EB? Until then: @edmondac  Can you just add that checksum and double-check with deleted archives?

**Comment by boegel** [comment]
@edmondac See fixed checksum in #9879

**Comment by edmondac** [comment]
@boegel that should do it, I think

**Comment by Flamefire** [comment]
Running test on our system now

**Comment by edmondac** [comment]
Test report by @bear-rsg
**SUCCESS**
Build succeeded for 11 out of 11 (8 easyconfigs in this PR)
bear-pg0212u15a.bear.cluster - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/10654d29e0daa0fc88eabfaaaad31e04 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
taurusi6325.taurus.hrsk.tu-dresden.de - Linux RHEL 7.7, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b75473ee2418bec25dc0812ecc6f655d for a full test report.

**Comment by edmondac** [comment]
OK @boegel we think we're happy with this :-)

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 7 out of 10 (8 easyconfigs in this PR)
taurusml30 - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
See https://gist.github.com/fa43829e112ba485821ce5bda243d52e for a full test report.

**Comment by terjekv** [comment]
> Test report by @Flamefire
> **FAILED**
> Build succeeded for 7 out of 10 (8 easyconfigs in this PR)
> taurusml30 - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
> See https://gist.github.com/fa43829e112ba485821ce5bda243d52e for a full test report.

Hm, an issue with Java/1.8 on power9?

```
== 2020-02-26 18:59:06,386 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): Module command 'module load Bazel/0.29.1-GCCcore-8.3.0' failed with exit code 1; stderr: Lmod hat den folgenden Fehler erkannt: Modul konnte nicht geladen werden: ""
     /sw/modules/ml/devel/Bazel/0.29.1-GCCcore-8.3.0.lua: /usr/share/lmod/lmod/libexec/MasterControl.lua:888: attempt to compare number with nil

Executing this command requires loading "Java/1.8" which failed while processing the following module(s):

    Module fullname             Module Filename
    ---------------             ---------------
    Bazel/0.29.1-GCCcore-8.3.0  /sw/modules/ml/devel/Bazel/0.29.1-GCCcore-8.3.0.lua
While processing the following module(s):
    Module fullname             Module Filename
    ---------------             ---------------
    Bazel/0.29.1-GCCcore-8.3.0  /sw/modules/ml/devel/Bazel/0.29.1-GCCcore-8.3.0.lua

; stdout: 
false
```

**Comment by Flamefire** [comment]
This seems to be an LMod 7(.8.9) issue where it can't find the proper module alias. Retrying with latest LMod

**Comment by terjekv** [comment]
Oh, yeah, right. That was brought up on Slack I think. Hopefully it's an easy LMod upgrade away from working.

**Comment by Flamefire** [comment]
Turns out to be a real LMod bug: https://github.com/TACC/Lmod/issues/436

I have a work-around and it's now starting to build TensorFlow. Going to take a while...

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
taurusml13 - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
See https://gist.github.com/6eb4e41b1b52aa648189fd405ca27a15 for a full test report.

**Comment by Flamefire** [comment]
Ok with the patches for R and the like (see my PRs) this works on Power9. So :+1: from me!

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 12 out of 12 (8 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/1c662b21205fa30e3eabb5b262012246 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (8 easyconfigs in this PR)
node3306.joltik.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, Python 3.6.8
See https://gist.github.com/bba1c53c820776a56561ae232c20040d for a full test report.

**Comment by boegel** [comment]
Going in, thanks @edmondac!

**Pull request merged**